pack: add support for bundles

ptaffet-jump opened the pull request.

ptaffet-jump committed:
pack: add support for bundles

mmcgee-jump committed:
bundle: add support for retrieving bundles from a block engine

palmerlao committed:
Update tuning page with 4 other tiles in fd_frankendancer.c

emwang-jump committed:
replay: streaming threaded poh tick verification

akhinvasara-jumptrading committed:
cleanup

small packet test

rm tag check in fd_quic_tx_buffered_raw

fix nonce gen

akhinvasara-jumptrading committed:
mv send_streams to used_streams if no new data to send

akhinvasara-jumptrading committed:
version outside loop

kbowers-jump committed:
fd_map_slot_para fixes

- Added missing function prototype when doing library build.

- Added stricter testing of parallel memo iteration and fixed rising
  edge / falling edge mixup bug uncovered in the process.

riptl committed:
Add objects to 'fdctl mem' output

riptl committed:
topo: detect invalid object align or footprint

emwang-jump committed:
replay: stub exec tile and recieve txns from replay

cali-jumptrading committed:
flamenco: implement move_precompile_verification_to_svm feature gate

anwayde committed:
build: remove 2 more machine types

lidatong committed:
ci: remove funk file from default-firedancer.toml

nbridge-jump commented on src/app/fdctl/run/tiles/fd_pack.c:
```diff
@@ -743,6 +776,16 @@ after_frag( fd_pack_ctx_t *     ctx,
+        int result = fd_pack_insert_bundle_fini( ctx->pack, ctx->current_bundle->bundle, ctx->current_bundle->txn_cnt, ctx->current_bundle->min_blockhash_slot, 0 );
+        insert_duration      += fd_tickcount();
+        ctx->insert_result[ result + FD_PACK_INSERT_RETVAL_OFF ]++;
+        fd_histf_sample( ctx->insert_duration, (ulong)insert_duration );
```
The non-bundle branch has this:
```
if( FD_LIKELY( result>=0 ) ) ctx->last_successful_insert = now;
```
Is this not needed in the bundle case?

kbhargava-jump committed:
flamenco: add asan to nightly

kbhargava-jump committed:
flamenco: update error on vm compute budget exceeded

kbhargava-jump committed:
flamenco: update builtin logic

kbhargava-jump committed:
nightly: add devnet builtin activation on boundary

riptl committed:
Reduce XDP_STATISTICS rate in net tile

The net tile previously polled XDP_STATISTICS every housekeeping
interval (~36µs with a typical fddev run!)

This commit increases the poll interval to 11ms.

Also replaces the clock_gettime() call that happens every
housekeep with fd_tickcount().

riptl committed:
Add 'fddev help' subcommand

Closes https://github.com/firedancer-io/firedancer/issues/1869

riptl committed:
fddev pktgen

Add packet generator tool for benchmarks

emwang-jump committed:
replay: fixes to tick verification for skipped slots and zero padding batches

ibhatt-jumptrading committed:
runtime: replacing scratch and wksp alloc with spad

mjain-jump committed:
fuzz, flamenco: fix executable account fetching bug for writable programdata accounts

kbhargava-jump committed:
flamenco: clean up syscall error handling

ibhatt-jumptrading committed:
flamenco: tpool and optimize epoch boundary
Co-authored-by: Manik Jain <mjain@jumptrading.com>

kbhargava-jump committed:
offline-replay: bump pages

riptl committed:
Add net_xsk_send_errors metric

riptl committed:
Net tile TX coalescing

anwayde committed:
add pre-commit check for trailing whitespaces

ibhatt-jumptrading committed:
flamenco: fixing allocator used for clock

kbhargava-jump committed:
nightly: bump pages and add thread mem bound

ibhatt-jumptrading committed:
flamenco: removing extraneous spad frame

ptaffet-jump committed:
resolv: fix two small bugs

yufeng-jump committed:
flamenco: various cpi fixes

Fix serialized owner accessor offset

Rework AccountInfo translation macros.  In particular, make sure that we match
Agave precisely in checking all of the following:
- Translation permission (r/w)
- Translation order
- Translation offsets, especially for the Rust boxes
- Translation length

Make sure that AccountInfo checks are gated by direct mapping feature flag in a
way that matches Agave

Move AccountInfo pubkey translations to earlier in the pipeline and cache the
results

Add InvalidArgument check to AccountMeta

Add more MissingAccount returns that match Agave

Signer translation should only be performed when length is non zero

Initialize instr_ctx in the instr stack

Remove translation of error codes in cpi harness

Clean up various error return handling

kbhargava-jump committed:
offline-replay: increase page allocation

riptl committed:
Level-triggered TX batch flush

Adds logic to flush the net tile TX queue when more than 70% of the
queue is occupied by packets added after the last flush op.

yufeng-jump committed:
flamenco: more cpi fixes

Use slice translator for byte array to allow for zero length translations

Implement W-permissioned translation on account info data length field

Re-translate with post length when updating caller

Log privilege escalation error messages in prepare_instruction

riptl committed:
Allow multiple input links for quic tile

mjain-jump committed:
features: tag features activated on all clusters

ptaffet-jump committed:
DEBUGGING ONLY: dump block engine keys

ptaffet-jump commented:
Rolled into #4114
